dvc/external_repo.py

```diff
@@ -106,7 +106,11 @@ def _pull_cached(self, out, path_info, dest):
             if out.changed_cache(filter_info=src):
                 self.cloud.pull(out.get_used_cache(filter_info=src))
 
-            failed = out.checkout(filter_info=src)
+            failed = False
+            try:
+                out.checkout(filter_info=src)
+            except CheckoutError:
+                failed = True
```

Comment:
Looks like there is an error here introduced earlier, one shouldn't move/remove if there is already an error. Simply raise `FileNotFoundError` here instead of using `failed` flag.